x/wasm/internal/keeper/keeper.go

```diff
@@ -1034,3 +1028,31 @@ func gasMeter(ctx sdk.Context) MultipiedGasMeter {
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
+
+// CoinTransferrer replicates the cosmos-sdk behaviour as in
+// https://github.com/cosmos/cosmos-sdk/blob/v0.41.4/x/bank/keeper/msg_server.go#L26
+type CoinTransferrer struct {
+	keeper types.BankKeeper
+}
+
+func NewBankCoinTransferrer(keeper types.BankKeeper) CoinTransferrer {
+	return CoinTransferrer{
+		keeper: keeper,
+	}
+}
+
+// TransferCoins transfers coins from source to destination account when coin send was enabled for them and the recipient
+// is not in the blocked address list.
+func (c CoinTransferrer) TransferCoins(ctx sdk.Context, fromAddr sdk.AccAddress, toAddr sdk.AccAddress, amt sdk.Coins) error {
```

Comment:
Thanks but this might be over engineered. Keep challenging me.  :-)
Main motivation was to not get into the type conversions for address and context (yet).